pages/curriculum/[lessonSlug]/mentor/index.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useMemo, useState } from 'react'
+import {
+  AddExerciseMutation,
+  GetAppProps,
+  Module,
+  useAddExerciseMutation,
+  withGetApp
+} from '../../../../graphql'
+import { useRouter } from 'next/router'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import { DropdownMenu } from '../../../../components/DropdownMenu'
+import { FormCard, MD_INPUT } from '../../../../components/FormCard'
+import { formChange } from '../../../../helpers/formChange'
+import { exercisesValidation } from '../../../../helpers/formValidation'
+import ExercisePreview from '../../../../components/ExercisePreview'
+import styles from '../../../../scss/mentorPage.module.scss'
+import { get } from 'lodash'
+import { ApolloError } from '@apollo/client'
+import QueryInfo from '../../../../components/QueryInfo'
+import { errorCheckAllFields } from '../../../../helpers/admin/adminHelpers'
+import * as Sentry from '@sentry/nextjs'
+
+type HeaderProps<T> = {
+  lesson: T
```

Comment:
Why is the `lesson` property type `T`? Is it because there is more than one lesson type we need to use?
If so, then I think it might be more type-safe to use something like `HeaderProps<T extends { title: string, modules: DetachedModule[] }>` where `type DetachedModule = Omit<Module, 'lesson' | 'author'>`, this will give an error at compile time if a value is passed in which doesn't have a title and modules property.

Reply:
Great idea! I'll try it.